### ISSUE TITLE
Machine Learning Modeling

### ISSUE THREAD

**Issue by levantuanem** [levantuanem/DS_Job_Recommend_Projects#4]
## Objective

Build, train, evaluate, and compare machine learning models for job-level prediction.

## Tasks

- [ ] Define X and y
- [ ] Create train/validation/test strategy
- [ ] Build baseline model
- [ ] Train multiple candidate models
- [ ] Select appropriate loss function
- [ ] Train models with appropriate objective/loss
- [ ] Evaluate model performance
- [ ] Compare models
- [ ] Tune hyperparameters
- [ ] Analyze bias
- [ ] Analyze variance
- [ ] Detect underfitting
- [ ] Detect overfitting
- [ ] Apply appropriate regularization where necessary
- [ ] Analyze feature importance
- [ ] Save the best model
- [ ] Document model selection

## Evaluation

Evaluate using appropriate metrics such as:

- Accuracy
- Precision
- Recall
- F1-score
- Confusion matrix

For multiclass classification, report performance for individual classes where appropriate.

## Output

- Baseline models
- Candidate models
- Hyperparameter tuning results
- Model evaluation results
- Best trained model
- Model comparison
- Bias/variance analysis

Recommended location:

`src/model/`

## Related Branch

`feature/model`
